src/main/kotlin/org/rust/lang/refactoring/generateConstructor/ui.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.refactoring.generateConstructor
+
+import com.intellij.codeInsight.generation.ClassMember
+import com.intellij.codeInsight.generation.MemberChooserObject
+import com.intellij.codeInsight.generation.MemberChooserObjectBase
+import com.intellij.ide.util.MemberChooser
+import com.intellij.openapi.project.Project
+import com.intellij.ui.SimpleColoredComponent
+import org.jetbrains.annotations.TestOnly
+import org.rust.lang.core.psi.RsStructItem
+import org.rust.openapiext.isUnitTestMode
+import javax.swing.JTree
+
+class RsStructMemberChooserObject(val base: MemberChooserObjectBase, val member: ConstructorArguments) : ClassMember {
+    private val text: String = "${member.argumentIdentifier}: ${member.type?.text ?: "()"}"
+    override fun renderTreeNode(component: SimpleColoredComponent?, tree: JTree?) {
+        component?.append(text)
+        component?.icon = member.type?.getIcon(0)
+    }
+
+    override fun getParentNodeDelegate(): MemberChooserObject? = base
+
+    override fun getText() = text
+
+    override fun equals(other: Any?): Boolean =
+        member == (other as? RsStructMemberChooserObject)?.member
+
+    override fun hashCode() = text.hashCode()
+}
+
+fun showConstructorArgumentsChooser(
+    structItem: RsStructItem,
+    project: Project
+): List<ConstructorArguments>? {
+
+    val base = MemberChooserObjectBase(structItem.name, structItem.getIcon(0))
+
+    val chooser = if (isUnitTestMode) {
+        MOCK!!
+    } else memberChooserDialog
+    return chooser(project,
+        ConstructorArguments.fromStruct(structItem).map { RsStructMemberChooserObject(base, it) })?.map { it.member }
```

Comment:
Let's extract `ConstructorArguments.fromStruct(structItem).map { RsStructMemberChooserObject(base, it) })?.map { it.member }`

Reply:
Also, maybe format it like this
```
    val base = MemberChooserObjectBase(structItem.name, structItem.getIcon(0))
    val chooser = if (isUnitTestMode) MOCK!! else memberChooserDialog
    val arguments = ConstructorArguments.fromStruct(structItem).map { RsStructMemberChooserObject(base, it) }
    return chooser(project, arguments)?.map { it.member }
```